core/src/main/java/org/mule/runtime/core/api/management/stats/FlowConstructStatistics.java

```diff
@@ -100,4 +100,45 @@ default void incConnectionErrors() {}
   default long getConnectionErrors() {
     return 0;
   }
+
+  /**
+   * Provides a counter for {@link #getExecutionErrors() execution errors} that is not affected by calls to {@link #clear()} or
+   * {@link ResetOnQueryCounter#getAndReset()} calls to other instances of {@link ResetOnQueryCounter}.
+   * 
+   * @return a counter for {@link #getExecutionErrors()}.
+   * 
+   * @since 4.5
+   */
+  ResetOnQueryCounter getExecutionErrorsCounter();
```

Comment:
as a consumer, I would expect this counter to be initialized with the real counter's current value, or at the very least, have the option to.

Looking at `DefaultResetOnQueryCounter` it seems like this counter always starts at zero, which generates a race-condition between the agent and the first processed messages